Core/Src/main.c

```diff
@@ -65,6 +68,18 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
+  const uint16_t COUNTER_PERIOD = 60000;
+  const uint16_t ADC_MAX_VALUE = 1023;
+  const uint8_t RX_BIT_MASK[3] = {0x0, 0x7F, 0xE0};
```

Comment:
I think you may be mis understanding how the data is coming in. SO, you do have right about the fact that 3 bytes are coming in, so that’s good! So it makes sense that you have 3 bit masks here.

It is the values of the masks that I am slightly confused about. So, if we look at the image i attached below, we see that the first byte of data that comes in (all the way to the left) are all `?` or “Don’t Care”. Which means that we don’t want any of that data, so having 0x00 as the first mask is technically correct but unnecessary because we wouldn’t even have to use that first byte of data.

NOW, the second byte, but the looks of it it contains B9 and B8 in the first and second bit location. This means that we can throw out the rest after that, so the second mask should be 0b00000011 in binary right (I will let you convert that to hex)?

Finally, the first byte contains ALL useful info, in specific, B7 to B0, so technically we don’t have to mask at all, but if you were to, since we want all the data, it would be 0b1111 1111.

Please let me know if that makes sense, if not, we can chat about it more! That being said, you may have to change line 129 accordingly.

<img width="1046" alt="image" src="https://github.com/UWARG/embedded-bootcamp/assets/69175242/01460534-fb0f-4935-9ab0-f2247eb1f5c4">

Reply:
That makes sense, I was sending my start bit earlier in the first byte which is why I had the weird masking. I still got the timing wrong though so this way is easier.